### DISCUSSION TITLE
images on oled 128x64 are displayed split in two parts with the beginning on the right side of the screen

### DISCUSSION THREAD

**Discussion by siliconchris1973** [micropython/micropython#9819] [Hardware & Peripherals] [answered]
Hi all, 

I am fairly new to micropython but already I am a big fan. So excuse me, if the following question is stupid.

I have one of those small OLED displays with 128 by 64 px (https://www.az-delivery.de/products/0-96zolldisplay)

I attached it to the PiCo according to documentation and I am able to display text. The issue is with images, albeit being displayed, they are actually displayed wrong. The images are displayed not from position 0 on the left but far more to the right (I'd say something around 60px) and then the remaining parts of the images are brought to the left - see attached pictures

![Oled-display-wrong](https://user-images.githubusercontent.com/7114193/199450065-38efa190-c69f-489e-876a-c582addefa74.jpg)

The display is 128x64 and the images are resized to 128x64 as well. I created the images by taking a jpg of the famous steamboat willie animation and converted it to PBM via one of the many online converter tools

My code is very simple:
```

    
    from machine import Pin, I2C
    import framebuf
    from time import sleep
    import ssd1306
    
    oled_width = 128
    oled_height = 64
    img_width = 128
    img_height = 64
    scl_pin = 17
    sda_pin = 16
    
    i2c = I2C(0, sda=Pin(sda_pin), scl=Pin(scl_pin))
    display = ssd1306.SSD1306_I2C(oled_width, oled_height, i2c)
    
    def getImageBuffer(fileToShow='sboat_00.pbm'):
        fto = 'images/' + fileToShow
        with open(fto, 'rb') as f:
            byte_array=bytearray(f.read())
        return(byte_array)
    
    def main():
        display.fill(0)
        data = framebuf.FrameBuffer(getImageBuffer('boat_00.pbm'), img_width, img_height, framebuf.MONO_HLSB)
        display.invert(1)
        display.blit(data, 0, 0)
        display.show()
        sleep(1)
        
        display.invert(0)
        display.poweroff()

    if __name__ == '__main__':
        main()


```
Please can someone tell me what I am doing wrong here?

kind regards,

Christian

**Comment by fdufnews** [comment, marked as answer]
The file starts with a header that you should skip.
Look here [https://netpbm.sourceforge.net/doc/pbm.html](https://netpbm.sourceforge.net/doc/pbm.html)

**Reply by siliconchris1973** [reply]
I followed your advice. Funny thing is - If I indeed skip the first 3 lines, my images are completely garbled over the screen. So I skipped only 2 lines of the PBM and this looks good. Another sequence I have, one with a jumping horse, indeed needs three lines of the PBM file to be skipped. Nevertheless, Thank you very much

**Reply by fdufnews** [reply]
The header has not a fixed size
You can have a comment just after a whitespace and starting with a #. All "fields" end with a "white space" character either of space, CR, LF, TAB, VT, or FF
In this example, with a comment, the raster image starts after the third 0x0A.
![Capture d’écran du 2022-11-02 16-50-50](https://user-images.githubusercontent.com/4077715/199537805-6d9b455f-54ee-47fd-b999-57b5e2cc5f0a.png)
